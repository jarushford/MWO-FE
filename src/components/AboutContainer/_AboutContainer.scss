.about-photo-container {
  height: 700px;

  h3 {
    color: white;
    font-size: 1.5rem;
    text-align: center;
    margin: 5px auto;
  }

  h4 {
    color: white;
    font-size: 1rem;
    text-align: center;
    margin-top: 8px;
  }
}

.photo {
  height: 500px;
  width: 350px;
  display: flex;
  flex-direction: column-reverse;
  border: 6px solid black;
  outline: 4px solid $purple;

  p {
    background: rgba(0, 0, 0, 0.7);
    border-radius: 7px;
    transform: translateY(20px);
    opacity: 0;
    transition: .7s ease;
  }

  i {
    font-size: 4rem;
    color: $purple;
    background: rgba(255, 255, 255, 0.7);
    padding: 0px 13.5px 5px;
    border-radius: 50%;
    cursor: pointer;
    transition: .7s ease;

    &:hover {
      opacity: 0
    }

    &:hover ~ p {
      opacity: 1;
      transform: translateY(0);
    }
  }
}