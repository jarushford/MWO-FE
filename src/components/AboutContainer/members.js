export const members = {
  Sabetta: {
    name: 'Jake Sabetta',
    inst: 'Guitar / Vocals',
    img: './assets/sabetta.jpg',
    bio: 'Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.'
  },
  Lauer: {
    name: 'Jake Lauer',
    inst: 'Drums',
    img: './assets/lauer.jpg',
    bio: '"Disco" Jake Lauer is a frisky lad. Not one to shy away from insulting the playing of another band member, tactfulness is not his strong suit. His honesty, however, does have its benefits; if you are trying to clean up for a night on the town, getting a Lauer seal of approval guarantees you are looking damn fine. Jake also dabbles in Mario Kart: Double Dash and loves to vacuum. If Mad Wallace were a rhinoceros, Lauer would be the horn.'
  },
  Jamie: {
    name: 'Jamie Rushford',
    inst: 'Guitar / Vocals',
    img: './assets/jamie.jpg',
    bio: 'Jamie Rushford is a man of few words. His stoic nature and piercing gaze were bestowed upon him at birth by Antheia, the Greek goddess of flowery wreaths. Not one to waste time idly, Jamie has made use of his laser focus, becoming a magnificent climber of the rocks, as well as a Level 13 master picking wizard. Calm, cool, and collected, his sense of style is timeless. If Mad Wallace were a rabbit, Jamie would be the eyes.'
  },
  David: {
    name: 'David Becker',
    inst: 'Bass / Vocals',
    img: './assets/david.jpg',
    bio: 'David Becker once lifted a pickup truck 2 feet into the air with only his right arm to retrieve a dying squirrel from underneath. The only thing stronger than David\'s massive shoulders is his tender heart. He would later resuscitate that squirrel with CPR techniques. David\'s easy-going personality meshes well with just about any human. David is also a master rock-climber. If Mad Wallace were a tortoise, David would be the shell.'
  }
}