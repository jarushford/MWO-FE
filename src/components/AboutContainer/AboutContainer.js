import React from 'react'
import { members } from './members'
import '../../main.scss'

export default function AboutContainer({ name }) {
  const memberCopy = members[name]

  return (
    <div className="about-photo-container">
      <div className="photo" style={{ background: `url(${memberCopy.image})` }}>
        <p>{memberCopy.bio}</p>
        
      </div>
      <h3>{memberCopy.name}</h3>
      <h4>{memberCopy.inst}</h4>
    </div>
  )
}