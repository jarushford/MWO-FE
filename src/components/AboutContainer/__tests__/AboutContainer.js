import React from 'react'
import { shallow } from 'enzyme'
import AboutContainer from '../AboutContainer'

describe('AboutConatiner', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<AboutContainer name="Sabetta" />)

    expect(wrapper).toMatchSnapshot()
  })
})