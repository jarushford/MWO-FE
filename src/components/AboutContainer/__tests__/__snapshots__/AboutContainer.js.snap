// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutConatiner should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutContainer
    name="Sabetta"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="photo"
          style={
            Object {
              "background": "url('./assets/sabetta.jpg')",
              "backgroundPosition": "center",
              "backgroundSize": "cover",
            }
          }
        >
          <i
            className="fas fa-angle-up"
          />
          <p>
            Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.
          </p>
        </div>,
        <h3>
          Jake Sabetta
        </h3>,
        <h4>
          Guitar / Vocals
        </h4>,
      ],
      "className": "about-photo-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-angle-up"
            />,
            <p>
              Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.
            </p>,
          ],
          "className": "photo",
          "style": Object {
            "background": "url('./assets/sabetta.jpg')",
            "backgroundPosition": "center",
            "backgroundSize": "cover",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-angle-up",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.",
            },
            "ref": null,
            "rendered": "Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jake Sabetta",
        },
        "ref": null,
        "rendered": "Jake Sabetta",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Guitar / Vocals",
        },
        "ref": null,
        "rendered": "Guitar / Vocals",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="photo"
            style={
              Object {
                "background": "url('./assets/sabetta.jpg')",
                "backgroundPosition": "center",
                "backgroundSize": "cover",
              }
            }
          >
            <i
              className="fas fa-angle-up"
            />
            <p>
              Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.
            </p>
          </div>,
          <h3>
            Jake Sabetta
          </h3>,
          <h4>
            Guitar / Vocals
          </h4>,
        ],
        "className": "about-photo-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-angle-up"
              />,
              <p>
                Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.
              </p>,
            ],
            "className": "photo",
            "style": Object {
              "background": "url('./assets/sabetta.jpg')",
              "backgroundPosition": "center",
              "backgroundSize": "cover",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-angle-up",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.",
              },
              "ref": null,
              "rendered": "Jake Sabetta is a passionate creature. His Italian blood runs hot, and that sultry emotion can be felt in every song he writes. His other lesser-known talent is in the kitchen. If ever in the mood for beans and rice, rice and beans, or beans and rice with some other stuff thrown in, Jake Sabetta is your man. He may not have as wide a breadth of dishes as Mario Batali, but his expertise with the simple things is unrivalled. If Mad Wallace were a lion, Sabetta would be the mane.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jake Sabetta",
          },
          "ref": null,
          "rendered": "Jake Sabetta",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Guitar / Vocals",
          },
          "ref": null,
          "rendered": "Guitar / Vocals",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
